metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/SummaryStatisticsFunctions.java

```diff
@@ -0,0 +1,198 @@
+/*
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.apache.metron.common.dsl.functions;
+
+import org.apache.commons.math3.stat.descriptive.SummaryStatistics;
+import org.apache.metron.common.dsl.BaseStellarFunction;
+
+import java.util.List;
+
+import static org.apache.metron.common.utils.ConversionUtils.convert;
+
+/**
+ * Provides Stellar functions that can calculate summary statistics on
+ * streams of data.
+ *
+ * These functions are limited to those that can be calculated in a
+ * single-pass so that the values are not stored in memory.  This leverages
+ * the commons-math SummaryStatistics class.
+ */
+public class SummaryStatisticsFunctions {
+
+  /**
+   * Initialize the summary statistics.
+   *
+   *  STATS_INIT ()
+   */
+  public static class Init extends BaseStellarFunction {
+    @Override
+    public Object apply(List<Object> args) {
+      return new SummaryStatistics();
+    }
+  }
+
+  /**
+   * Add an input value to those that are used to calculate the summary statistics.
+   *
+   *  STATS_ADD (value, stats)
+   */
+  public static class Add extends BaseStellarFunction {
+    @Override
+    public Object apply(List<Object> args) {
+      double value = convert(args.get(0), Double.class);
+      SummaryStatistics stats = convert(args.get(1), SummaryStatistics.class);
+      stats.addValue(value);
+      return null;
```

Comment:
Yeah, let's not do that.  Instead, let's make it so that the stats are first and the user can pass as many nums as they like, so that would be `STATS_ADD(s, 1, 2, 3)`